mindone/models/t5/t5.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: utf-8 -*-
+import html
+import logging
+import re
+import urllib.parse as ul
+
+import ftfy
+from bs4 import BeautifulSoup
+
+import mindspore as ms
+from mindspore import Tensor, nn
+
+from mindone.models.t5.t5_modeling import get_t5_encoder, get_t5_tokenizer
+
+logger = logging.getLogger(__name__)
+
+
+class T5Embedder(nn.Cell):
+    available_models = ["t5-v1_1-xxl"]
+    bad_punct_regex = re.compile(
+        r"["
+        + "#®•©™&@·º½¾¿¡§~"
+        + r"\)"
+        + r"\("
+        + r"\]"
+        + r"\["
+        + r"\}"
+        + r"\{"
+        + r"\|"
+        + r"\\"
+        + r"\/"
+        + r"\*"
+        + r"]{1,}"
+    )  # noqa
+
+    def __init__(
+        self,
+        cache_dir,
+        use_text_preprocessing=True,
+        model_max_length=120,
+        pretrained_ckpt=None,
```

Comment:
Since this code will be added to the library, please include type hints.

Reply:
And docstrings for APIs, thanks